core/scripts/vrfv2/testnet/main.go

```diff
@@ -123,6 +126,72 @@ func main() {
 		tx, err := batchBHS.StoreVerifyHeader(owner, blockRange, rlpHeaders)
 		helpers.PanicErr(err)
 		fmt.Println("storeVerifyHeader(", blockRange, ", ...) tx:", helpers.ExplorerLink(chainID, tx.Hash()))
+	case "batch-bhs-backwards":
+		cmd := flag.NewFlagSet("batch-bhs-backwards", flag.ExitOnError)
+		batchAddr := cmd.String("batch-bhs-address", "", "address of the batch bhs contract")
+		startBlock := cmd.Int64("start-block", -1, "block number to start from. Must be in the BHS already.")
+		endBlock := cmd.Int64("end-block", -1, "block number to end at. Must be less than startBlock")
+		batchSize := cmd.Int64("batch-size", -1, "batch size")
+		gasMultiplier := cmd.Int64("gas-price-multiplier", 1, "gas price multiplier to use, defaults to 1 (no multiplication)")
+		helpers.ParseArgs(cmd, os.Args[2:], "batch-bhs-address", "start-block", "end-block", "batch-size")
+
+		batchBHSABI, err := batch_blockhash_store.BatchBlockhashStoreMetaData.GetAbi()
+		helpers.PanicErr(err)
+
+		blockRange, err := decreasingBlockRange(big.NewInt(*startBlock-1), big.NewInt(*endBlock))
+		helpers.PanicErr(err)
+
+		batchbhsaddr := common.HexToAddress(*batchAddr)
+
+		for i := 0; i < len(blockRange); i += int(*batchSize) {
+			j := i + int(*batchSize)
+			if j > len(blockRange) {
+				j = len(blockRange)
+			}
+
+			blockNumbers := blockRange[i:j]
+			blockHeaders, err := getRlpHeaders(ec, blockNumbers)
+			fmt.Println("storing blockNumbers:", blockNumbers)
+			helpers.PanicErr(err)
+
+			payload, err := batchBHSABI.Pack("storeVerifyHeader", blockNumbers, blockHeaders)
+			helpers.PanicErr(err)
+
+			nonce, err := ec.PendingNonceAt(context.Background(), owner.From)
+			helpers.PanicErr(err)
+
+			fmt.Println("using nonce:", nonce)
+
+			fmt.Println("estimating gas")
+
+			gasLimit, err := ec.EstimateGas(context.Background(), ethereum.CallMsg{
```

Comment:
Why is gas estimation + using the generic SendTransaction necessary vs just calling the function on the typed batched_bhs golang type?

Reply:
Ah good point. I think initially I was going to try bumping the nonce manually and sending e.g 5 transactions at a time but I think that's a bit more error prone without the BPTXM. Will switch to using the go wrapper.

Reply:
You should be able to set the nonce manually with the go wrapper too by setting the owner.Nonce field